packages/number-field/src/NumberField.ts

```diff
@@ -402,7 +403,27 @@ export class NumberField extends TextfieldBase {
         super.handleChange();
     }
 
-    protected override handleInput(): void {
+    protected handleCompositionStart(): void {
+        this.isComposing = true;
+    }
+
+    protected handleCompositionEnd(): void {
+        this.isComposing = false;
+        requestAnimationFrame(() => {
+            this.inputElement.dispatchEvent(
```

Comment:
I wish I remembered why I did this in the `rAF()`. It's the only part that seems clearly brittle/directly working to not step on the OS's toes, which means easy to break.